### PR TITLE
fix: allow setting a list as a field value

### PR DESCRIPTION
**Pull request by JBthePenguin**
To set form data, use the `setlist` method of `QueryDict` when the provided value is `Iterable`.

### CODE REVIEW

**Comment by GitRon** [comment]
Hi @JBthePenguin 

the code looks good. Would you mind adding at least one test for the changes?

Best regards  
Ronny

**Comment by JBthePenguin** [comment]
Hi @GitRon !
OK, I will add a test asap. The goal is to have 100% coverage, right?

**Comment by GitRon** [comment]
@JBthePenguin That would be awesome. But having tests to ensure the new functionality works and keeps working is important, too 🙂

**Comment by JBthePenguin** [comment]
@GitRon Test added.

File: django_dynamic_admin_forms/admin.py [hunk 1]
```diff
@@ -59,7 +60,10 @@ def render_field(request) -> HttpResponse:
 
                 bound_field.field.queryset = queryset
                 bound_field.form.data = bound_field.form.data.copy()
-                bound_field.form.data[field_name] = value
+                if isinstance(value, Iterable):
```
Comment (GitRon) [inline django_dynamic_admin_forms/admin.py]: issue: I've asked the AI for a second opinion. Seems legit, doesn't it?

> Potential Issue: str is Iterable
> 
> In Python, str is an Iterable. If a get_dynamic_*_field method returns a string as value, it will be passed to .setlist(), which would split the string into individual characters — almost certainly not the intended behavior.
> 
>   For example, if value = "hello", then setlist(field_name, "hello") would set the field to ["h", "e", "l", "l", "o"].
> 
>   The fix should explicitly exclude strings:
> 
>   if isinstance(value, Iterable) and not isinstance(value, str):
> 
>   Or alternatively, check specifically for list/tuple:
> 
>   if isinstance(value, (list, tuple)):

File: tests/test_admin_unit.py [hunk 2]
```diff
@@ -80,3 +80,18 @@ def test_render_field_with_permission_denied(self):
         # You can adjust the permissions for the user to raise PermissionDenied
         with self.assertRaises(PermissionDenied):
             mixin.render_field(request)
+
+    def test_render_field_with_custom_method_for_many_to_many_field(self):
```
Comment (GitRon) [inline tests/test_admin_unit.py:85]: issue: Another valid AI suggestin IMHO

>  The new test (test_render_field_with_custom_method_for_many_to_many_field) verifies the happy path returns a 200 status. It's a reasonable integration-style test, but could be strengthened by:
> 
>   1. Asserting the response body contains the expected widget HTML with the pre-selected values ("1" and "2").
>   2. Adding a test that a non-list value (e.g., a string) still works correctly after this change — to catch the str is Iterable edge case.

**Review by GitRon — COMMENTED** [review]
Hi @JBthePenguin! Thanks for the tests. I've asked the AI for a second opinion since I've took over this package and I'm not the one that built it in the first place.

What do you think?

File: tests/test_admin_unit.py (hunk 2 above)
Comment (JBthePenguin) [reply tests/test_admin_unit.py:85]: @GitRon Thanks for comments! That seems legit, yes. I will post an update as soon as I find a moment to do so.

Comment (GitRon) [reply tests/test_admin_unit.py:85]: You're the best ❤️

File: django_dynamic_admin_forms/admin.py (hunk 1 above)
Comment (JBthePenguin) [reply django_dynamic_admin_forms/admin.py]: @GitRon Fix done.

File: tests/test_admin_unit.py (hunk 2 above)
Comment (JBthePenguin) [reply tests/test_admin_unit.py:85]: @GitRon Two tests added. The first one to check if `setlist` method is called when value is a `list`, and the second one to check if `setlist` method is not called when value is a `str`.

Comment (GitRon) [reply tests/test_admin_unit.py:85]: Thanks! ❤️  Will have a look once I find a minute 😅

**Review by GitRon — APPROVED** [review]
Great, thanks! 👌

**Comment by GitRon** [comment]
@JBthePenguin Thanks for the work! Just published the new release: https://pypi.org/project/django-dynamic-admin-forms

**Comment by JBthePenguin** [comment]
@GitRon Thanks!
